maps/rivers_and_bridges.go

```diff
@@ -0,0 +1,386 @@
+package maps
+
+import (
+	"github.com/BattlesnakeOfficial/rules"
+)
+
+func init() {
+	globalRegistry.RegisterMap("hz_rivers_bridges", RiverAndBridgesMediumHazardsMap{})
+	globalRegistry.RegisterMap("hz_rivers_bridges_lg", RiverAndBridgesLargeHazardsMap{})
+	globalRegistry.RegisterMap("hz_rivers_bridges_xl", RiverAndBridgesExtraLargeHazardsMap{})
+}
+
+func getUnoccupiedPoints(lastBoardState *rules.BoardState) []rules.Point {
```

Comment:
I think it's a little confusing having two similarly named functions in the rules and maps packages, where one actually wraps the other. Maybe either:
- rename to `getUnoccupiedPointsWithHazards`
- add a boolean param to `rules.GetUnoccupiedPoints` that makes it exclude hazards as well, and use that instead?

Reply:
I'm open to either option.  Feels like we may want to do this more often so maybe #2 makes the most sense?

Reply:
I think we want to move towards providing more reusable utilities for maps to do common operations like this, so 2 is probably better.

Reply:
ok, i'll take a swing at this and see if I can figure out how rules.GetUnoccupiedPoints() works